File: web/src/services/apiClient.ts
const apiBaseUrl = import.meta.env.VITE_API_URL ?? "http://localhost:5000";

export async function apiRequest<T>(path: string, options?: RequestInit): Promise<T> {
	const response = await fetch(`${apiBaseUrl}${path}`, { headers: { "Content-Type": "application/json", ...(options?.headers ?? {}) }, ...options });
	if (!response.ok) { const error = await response.json().catch(() => ({ message: "Request failed" })); throw new Error(error.message ?? "Request failed"); }
	const body = await response.json(); return body.data ?? body;
}

File: mobile/lib/core/network/api_client.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class ApiClient {
	ApiClient({String? baseUrl}) : baseUrl = baseUrl ?? const String.fromEnvironment('API_URL', defaultValue: 'http://10.0.2.2:5000');
	final String baseUrl;

	Future<dynamic> request(String path, {String method = 'GET', Map<String, dynamic>? body}) async {
		final uri = Uri.parse('$baseUrl$path');
		final response = method == 'POST' ? await http.post(uri, headers: {'Content-Type': 'application/json'}, body: jsonEncode(body)) : await http.get(uri);
		final decoded = jsonDecode(response.body);
		if (response.statusCode < 200 || response.statusCode >= 300) throw Exception(decoded['message'] ?? 'Request failed');
		return decoded['data'] ?? decoded;
	}
}
